Sources/GRPCInterceptors/Tracing/ServerOTelTracingInterceptor.swift

```diff
@@ -15,24 +15,42 @@
  */
 
 public import GRPCCore
-internal import Tracing
+internal import Synchronization
+package import Tracing
 
 /// A server interceptor that extracts tracing information from the request.
 ///
 /// The extracted tracing information is made available to user code via the current `ServiceContext`.
+///
 /// For more information, refer to the documentation for `swift-distributed-tracing`.
-public struct ServerTracingInterceptor: ServerInterceptor {
+///
+/// This interceptor will also inject all required and recommended span and event attributes, and set span status, as defined by
+/// OpenTelemetry's documentation on:
+/// - https://opentelemetry.io/docs/specs/semconv/rpc/rpc-spans
+/// - https://opentelemetry.io/docs/specs/semconv/rpc/grpc/
+public struct ServerOTelTracingInterceptor: ServerInterceptor {
   private let extractor: ServerRequestExtractor
-  private let emitEventOnEachWrite: Bool
+  private let traceEachMessage: Bool
+  private var serverHostname: String
+  private var networkTransportMethod: String
 
-  /// Create a new instance of a ``ServerTracingInterceptor``.
+  /// Create a new instance of a ``ServerOTelTracingInterceptor``.
   ///
-  /// - Parameter emitEventOnEachWrite: If `true`, each response part sent and request part
-  /// received will be recorded as a separate event in a tracing span. Otherwise, only the request/response
-  /// start and end will be recorded as events.
-  public init(emitEventOnEachWrite: Bool = false) {
+  /// - Parameters:
+  ///  - severHostname: The hostname of the RPC server. This will be the value for the `server.address` attribute in spans.
+  ///  - networkTransportMethod: The transport in use (e.g. "tcp", "udp"). This will be the value for the
```

Comment:
nit: this is more applicable

```suggestion
  ///  - networkTransportMethod: The transport in use (e.g. "tcp", "unix"). This will be the value for the
```